Fix issue 1639 by adding project name to launch configuration

jonpasski opened the pull request:
First attempt at a pull request for issue #1639, based on the 3.3.0 REL tag.

gamerson commented:
hey @jonpasski  could you rebase on master and resend?

jonpasski commented:
@gamerson I'm a git noob and might need some hand holding. I fetched origin, checked out and pulled master, then tried to `git checkout issue1639-3_3_0_REL` onto master. This caused a bunch of merge conflicts in things I didn't touch, which doesn't seem right to me...

gamerson commented:
Understand.  Here is how I would do it.

git checkout master
git pull --ff-only origin master

now you are up2date.

Then create a whole new branch based on master.

git checkout -b issue1639fix

then cherry pick your change from previously created branch

git cherry-pick 7982376a29cccc304bd1185915a0f0f293bc26d6

This will likely have conflicts, but you can see the conflicts by doing

git status

Then you can fix up the conflicts because we have recently changed these
two classes that you modified in your patch.  So you will likely have to
look at those changes you made and try to 'reimplement' those changes on
the latest master.  Some of the logic may need to be adjusted.

On Fri, Jan 27, 2017 at 2:31 PM, Jon Passki <notifications@github.com>
wrote:

> @gamerson <https://github.com/gamerson> I'm a git noob and might need
> some hand holding. I fetched origin, checked out and pulled master, then
> tried to git checkout issue1639-3_3_0_REL onto master. This caused a
> bunch of merge conflicts in things I didn't touch, which doesn't seem right
> to me...
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/bndtools/bndtools/pull/1640#issuecomment-275767033>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAkVFXpaPncTX6j3Ys2FhN-koSzbpNDCks5rWlQ9gaJpZM4LwEH0>
> .
>



-- 
Greg Amerson
Liferay Developer Tools
Liferay, Inc. www.liferay.com

jonpasski commented:
Thanks @gamerson! That seemed to work, although a bit bumpy. I had 3.3.0-REL installed but it couldn't seem to build the `issue1639fix` branch, even after I manually merged changes via the conflict editor. So I just dropped to the command line, ran `./gradlew --continue`, got that built, then installed that version. After manually having to clean / build the API project and a couple others, all the projects built in the UI too.

I then went through the same steps as before:

* Installed the new version into the debugger instance.
* Deleted the previous launch config, launched a new one. (The newer version of bndtools seems to cause a bug in a local bundle, which throw an exception. However, the debugger still connected and it opened a gogo shell to the target Eclipse instance.)
* Edited the launch config, navigated to the Connect tab, noted the project name is there.
* Re-launched and was able to connect again. It also threw the same exception again: 

> org.osgi.framework.BundleException: Could not resolve module: org.example.provider [700]
>  Unresolved requirement: Require-Capability: osgi.extender; filter:="(&(osgi.extender=osgi.component)(version>=1.3.0)(!(version>=2.0.0)))"

I'm working w/ Equinox on Neon, which I think only supports 1.2.0, so I'm gonna read up on how to deal w/ the above after this :)

So outside of the above exception, seems to be working as it did before. Travis shows it as [being built successfully](https://travis-ci.org/jonpasski/bndtools/builds/196018449). Should I pull request that branch instead now?

bjhargrave commented:
The commits in the PR can't be used. You can either create a new PR with the single commit, or force push them here.

gamerson commented:
Yeah I would just send a whole new PR from your new branch with fixed commits and then you could close this one.

jonpasski commented:
What do you recommend BJ? I know how to do the former. I haven't done the prior before, although I'm happy to learn if you recommended it :)




Jon
> On Jan 27, 2017, at 15:10, BJ Hargrave <notifications@github.com> wrote:
> 
> The commits in the PR can't be used. You can either create a new PR with the single commit, or force push them here.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

bjhargrave commented:
Let's just do a new PR.

On Fri, Jan 27, 2017 at 15:42 Jon Passki <notifications@github.com> wrote:

> What do you recommend BJ? I know how to do the former. I haven't done the
> prior before, although I'm happy to learn if you recommended it :)
>
>
>
>
> Jon
> > On Jan 27, 2017, at 15:10, BJ Hargrave <notifications@github.com> wrote:
> >
> > The commits in the PR can't be used. You can either create a new PR with
> the single commit, or force push them here.
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub, or mute the thread.
> >
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/bndtools/bndtools/pull/1640#issuecomment-275804192>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAQ8sqHqSDw3Fj3MA5-0zTV2VL4760Ibks5rWoDPgaJpZM4LwEH0>
> .
>
-- 
BJ

jonpasski commented:
Opened PR #1641 and closing this.
